text/0000-let.md

```diff
@@ -0,0 +1,265 @@
+- Start Date: 2017-01-14
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+Introduce a `let` keyword to bind values to names within a handlebars
+template.
+
+Block form:
+
+```handlebars
+{{#let "Milk" "Cereal" as |one two|}}
+  Breakfast:
+  1. {{one}}
+  2. {{two}}
+{{/let}}
+```
+
+or inline:
+
+```handlebars
+{{let one="Milk" two="Cereal"}}
+
+Breakfast:
+1. {{one}}
+2. {{two}}
+```
+
+# Motivation
+
+The primary method for supplying named values to a template is via
+properties on its backing scope (either a controller or a
+component). These properties are specified in JavaScript which, while
+as a mechanism is very well understood by the community, also comes
+with several drawbacks.
+
+The first is that it spreads the requisite knowledge to understanding
+a template across two places. In other words, it is impossible to understand
+what exactly it does without simultaneously holding both the `.hbs`
+and `.js` files together and continuously merging them with your
+mind. The lack of this overhead is one of the commonly cited strengths
+of JSX over Handlebars.
+
+Another is that computed properties are error-prone because of their
+requirement to explicitly enumerate dependencies. If the computation
+of a property has incorrectly specified dependent keys it will break
+in mysterious ways that usually involve a sudden and hard-to-reproduce
+lack of reactivity to changing inputs.
```

Comment:
>I think a better approach to fix this is to make CPs easier. One way of doing so would be to come up with a special kind of CP that automatically tracks its dependencies. Doing that should (more or less) easily be possible, I just never found the time yet to build a prototype.

You can not do this without code that comprehends the JavaScript AST of the function representing your property. That would be very cool, but would also come with its own complexities. Handlebars templates already comprehend the dependency flow at a syntax level and so we can lean on them is free.